mainargs/src-3/Macros.scala

```diff
@@ -137,12 +139,9 @@ object Macros {
       }
     }
 
-    val base = Apply(fct, accesses.head)
-    val application: Apply = accesses.tail.foldLeft(base)((lhs, args) => Apply(lhs, args))
-    val expr = application.asExpr
-    expr
+    methodOwner.asTerm.select(method).appliedToArgss(accesses).asExpr
```

Comment:
we first select the method using its `Symbol` on the owner and apply all of its parameter lists using `.appliedToArgss` as opposed to handrolling this tree